README.md

```diff
@@ -296,33 +296,45 @@ Or:
+    * `accessAssignments`: A list that contains each of the access assignments that were applied, where each element's `usersAndRoles` property is a list of the users and roles and the `channels` property is a list of the channels that were granted to them. Note that, as per the sync function API, each role element's value is prefixed with "role:".
+  5. `onDocumentChannelAssignmentSucceeded`: Executed immediately after channels are assigned to the document. The last step before the sync function is finished executing and the document revision is written. The custom action metadata object parameter includes properties from all previous events in addition to the following properties:
+    * `documentChannels`: A list of channels that were assigned to the document.
 
-An example of an `onAuthorizationSucceeded` custom action:
+An example of an `onAuthorizationSucceeded` custom action that stores a property in the metadata object parameter for later use by the `onDocumentChannelAssignmentSucceeded` custom action:
 
 ```
     customActions: {
-      onAuthorizationSucceeded: function(doc, oldDoc) {
+      onAuthorizationSucceeded: function(doc, oldDoc, customActionMetadata) {
+        var extraChannel = customActionMetadata.documentTypeId + '-modify';
         if (oldDoc && !oldDoc._deleted) {
-          // If the document is being replaced or deleted, ensure the user has the document's "-modify" channel in addition to one of the
-          // channels defined in the document definition's "channels" property
-          requireAccess(doc._id + '-modify');
+          // If the document is being replaced or deleted, ensure the user has the document type's "-modify" channel in addition to one of
+          // the channels from the document definition's "channels" property that was already authorized
+          requireAccess(extraChannel);
         }
+
+        // Store the extra modification validation channel name for future use
+        customActionMetadata.extraModifyChannel = extraChannel;
+      },
+      onDocumentChannelAssignmentSucceeded: function(doc, oldDoc, customActionMetadata) {
+        // Ensure the extra modification validation channel is also assigned to the document
+        channel(customActionMetadata.extraModifyChannel);
       }
     }
 ```
 
-* `allowAttachments`: (optional) Whether to allow the addition of [file attachments](http://developer.couchbase.com/documentation/mobile/current/develop/references/sync-gateway/rest-api/document-public/put-db-doc-attachment/index.html) for the document type. Defaults to `false` to prevent malicious/misbehaving clients from polluting the bucket/database with unwanted files.
```

Comment:
This block has not changed. It's simply been moved above `customActions` because the options are more likely to be useful to a general audience.